### PR TITLE
add verify enterprise contract task

### PR DESCRIPTION
**Pull request by joejstuart**
Adding verify enterprise contract task to be managed with other tasks belonging to dance/rhtap.

https://issues.redhat.com/browse/EC-429

### CODE REVIEW

File: task/verify-enterprise-contract/0.1/verify-enterprise-contract.yaml [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        fi
+
+        echo 'Initializing TUF root...'
+        cosign initialize --mirror "${TUF_MIRROR}" --root "${TUF_MIRROR}/root.json"
+        echo 'Done!'
+      env:
+        - name: TUF_MIRROR
+          value: "$(params.TUF_MIRROR)"
+    - name: validate
+      image: quay.io/enterprise-contract/ec-cli:snapshot
```
Comment (zregvart) [inline task/verify-enterprise-contract/0.1/verify-enterprise-contract.yaml:145]: We probably need to use `registry.redhat.io/rhtas-tech-preview/ec-rhel9:1.0.alpha` here and in the step below, or work on a exception [here](https://github.com/release-engineering/rhtap-ec-policy/blob/843e20385d4eaf9e61f8e6dad7845e6fb8248b2d/data/rule_data.yml#L10-L16)

**Review by zregvart — COMMENTED** [review]
I think we'll also need the help of the build team to setup a `task-verify-enterprise-contract` repository in [redhat-appstudio-tekton-catalog](https://quay.io/organization/redhat-appstudio-tekton-catalog)

File: task/verify-enterprise-contract/0.1/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+* **SSL_CERT_DIR**: Path to a directory containing SSL certs to be used when communicating
+        with external services.
+* **STRICT**: Fail the task if policy fails. Set to "false" to disable it.
+* **HOMEDIR**: Value for the HOME environment variable.
+* **EFFECTIVE_TIME**: Run policy checks with the provided time.
+
+
+## Usage
+
+This TaskRun runs the Task to verify an image. This assumes a policy is created and stored on the cluster with hte namespaced name of `enterprise-contract-service/default`. For more information on creating a policy, refer to the Enterprise Contract [documentation](https://enterprise-contract.github.io/ecc/main/index.html).
```
Comment (simonbaird) [inline task/verify-enterprise-contract/0.1/README.md:27]:
```suggestion
This TaskRun runs the Task to verify an image. This assumes a policy is created and stored on the cluster with the namespaced name of `enterprise-contract-service/default`. For more information on creating a policy, refer to the Enterprise Contract [documentation](https://enterprise-contract.github.io/ecc/main/index.html).
```

File: task/verify-enterprise-contract/0.1/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        namespace is omitted the namespace where the task runs is used. For inline policy provide the [specification](https://enterprise-contract.github.io/ecc/main/reference.html#k8s-api-github-com-enterprise-contract-enterprise-contract-controller-api-v1alpha1-enterprisecontractpolicyspec) as JSON.
+* **PUBLIC_KEY**: Public key used to verify signatures. Must be a valid k8s cosign
+        reference, e.g. k8s://my-space/my-secret where my-secret contains
+        the expected cosign.pub attribute.
+* **REKOR_HOST**: Rekor host for transparency log lookups
+* **SSL_CERT_DIR**: Path to a directory containing SSL certs to be used when communicating
+        with external services.
+* **STRICT**: Fail the task if policy fails. Set to "false" to disable it.
+* **HOMEDIR**: Value for the HOME environment variable.
+* **EFFECTIVE_TIME**: Run policy checks with the provided time.
```
Comment (simonbaird) [inline task/verify-enterprise-contract/0.1/README.md:22]: Nit pick: In what format? What does it do?

Comment (simonbaird) [reply task/verify-enterprise-contract/0.1/README.md:22]: Never mind, I see these match the param descriptions.

**Comment by simonbaird** [comment]
Should include a Jira ref in the commit message and/or the PR.

**Comment by joejstuart** [comment]
> I think we'll also need the help of the build team to setup a `task-verify-enterprise-contract` repository in [redhat-appstudio-tekton-catalog](https://quay.io/organization/redhat-appstudio-tekton-catalog)

@jduimovich can you help with this?

File: task/verify-enterprise-contract/0.1/verify-enterprise-contract.yaml (hunk 1 above)
Comment (joejstuart) [reply task/verify-enterprise-contract/0.1/verify-enterprise-contract.yaml]: yes, good catch.

**Review by zregvart — APPROVED** [review]

**Review by jduimovich — APPROVED** [review]

**Pull request merged**